File: erlang/armstrong-numbers/src/armstrong_numbers.erl
-module(armstrong_numbers).

-export([is_armstrong_number/1]).

power_number([],_Exp,Acc) ->
    Acc;
power_number([Head | Tail],Exp,Acc) ->
    Digit = list_to_integer([Head]),
    Power = trunc(math:pow(Digit,Exp)),
    power_number(Tail,Exp,[Power | Acc]).

is_armstrong_number(Number, Number) ->
    true;
is_armstrong_number(_Number, _Sum) ->
    false.

is_armstrong_number(Number) ->
    NumberList = integer_to_list(Number),
    PowerList = power_number(NumberList,length(NumberList),[]),
    is_armstrong_number(Number,lists:sum(PowerList)).

File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
  if (candidate == 0) {
    return true;
  }

  int acc = candidate / 10, d = candidate % 10, power = log10(candidate) + 1,
      sum = pow(d, power);
  while (acc != 0) {
    d = acc % 10;
    acc /= 10;
    sum += pow(d, power);
  }

  return sum == candidate;
}
